### PR TITLE
Routing

### LINKED ISSUE #19: Add client-side routing
For now we can have a home page at `/` with a list of links to board pages at `boards/:id`

Suggest we use @reach/router having read this https://reacttraining.com/blog/reach-react-router-future

### PR DESCRIPTION
**Pull request by benji6**
Closes #19

Also makes a start on #21

Quite a lot moved around in this but we have client-side routing now.

Home page:
screencapture-localhost-1234-2019-05-25-14_50_40

Board page:
screencapture-localhost-1234-boards-1-2019-05-25-14_50_50

### CODE REVIEW

File: src/app/components/App.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import Main from './Main';
+
+const client = new ApolloClient({
+  uri: `http://${window.location.hostname}:4000/graphql`,
+});
+
+export default function App() {
+  return (
+    <ApolloProvider client={client}>
+      <>
```
Comment (robcalcroft) [inline src/app/components/App.js]: Is a fragment needed here?

File: src/app/components/pages/NotFound.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { Redirect } from '@reach/router';
+import * as React from 'react';
```
Comment (robcalcroft) [inline src/app/components/pages/NotFound.js]: Typescript pattern?

Comment (benji6) [reply src/app/components/pages/NotFound.js]: My JS is out of practice! 😄

File: src/app/components/App.js (hunk 1 above)
Comment (benji6) [reply src/app/components/App.js]: apparently not - will remove!

File: src/api/model/boards.js [hunk 3]
```diff
@@ -1,21 +1,36 @@
 const pool = require('./pool');
 
+const formatRow = row => {
```
Comment (robcalcroft) [inline src/api/model/boards.js]: Could you test using the sql method of `camel_case as camelCase` and see how well it works, if we can offload this logic to sql would save us having this code and save on maintenance cost of this function. I think camelCase in postgres will need `camel_case as "camelCase"`

Comment (benji6) [reply src/api/model/boards.js]: I've pushed up a commit - let me know what you think. It's unfortunate you have to list out all the column names in SQL this way, but I think it's better to do the renaming higher up so I think this is nicer.

Also fixed a bug with GraphQL `deleteBoard` mutation always erroring

Comment (robcalcroft) [reply src/api/model/boards.js]: Sweet! Yeah you're right having to do `returning x, x, x` is unfortunate but yep i think this is the way to go 👍

**Review by robcalcroft — APPROVED** [review]

**Pull request merged**